src/main/java/seedu/duke/parsers/AddTaskParser.java

```diff
@@ -0,0 +1,107 @@
+package seedu.duke.parsers;
+
+import seedu.duke.commands.AddCommand;
+import seedu.duke.commands.Command;
+import seedu.duke.exceptions.ModHappyException;
+import seedu.duke.exceptions.GeneralParseException;
+import seedu.duke.exceptions.EmptyParamException;
+import seedu.duke.exceptions.MissingCompulsoryParameterException;
+import seedu.duke.exceptions.InvalidCompulsoryParameterException;
+import seedu.duke.util.StringConstants;
+
+import java.util.HashMap;
+import java.util.Objects;
+
+/**
+ * This Parser supports the "add" command.
+ */
+public class AddTaskParser extends Parser {
+    private static final String TASK_STR = StringConstants.TASK_STR;
+    private static final String TASK_DESCRIPTION_STR = StringConstants.TASK_DESCRIPTION_STR;
+    private static final String TASK_ESTIMATED_WORKING_TIME_STR = StringConstants.TASK_ESTIMATED_WORKING_TIME_STR;
+    private static final String TASK_NAME = StringConstants.TASK_NAME;
+    private static final String TASK_DESCRIPTION = StringConstants.TASK_DESCRIPTION;
+    private static final String TASK_WORKING_TIME = StringConstants.TASK_ESTIMATED_WORKING_TIME;
+    private static final String TASK_MODULE = StringConstants.TASK_MODULE;
+    private String userInput;
+
+    // Unescaped regex for testing (split across a few lines):
+    // (task\s+\"(?<taskName>[^\"]+)\"(\s+(-m\s+(?<taskModule>\w+)))?(\s+-d\s+\"(?<taskDescription>[^\"]+)\")?
+    // (\s+-t\s+\"(?<estimatedWorkingTime>[^\"]+)\")?)(?<invalid>.*)
+
+    /* Explanation for regex:
+     * (task\s+\"(?<taskName>[^\"]+)\"                   -- matches [task "taskName"].
+     * (\s+(-m\s+(?<taskModule>\w+)))?                   -- matches [-m taskModule] if present. Optional
+     *                                                      Note that taskModule does not require "", but must be a
+     *                                                      single word composed of [a-zA-Z0-9_].
+     * (\s+-d\s+\"(?<taskDescription>[^\"]+)\")?         -- matches [-d "taskDescription"] if present. Optional
+     * (\s+-t\s+\"(?<estimatedWorkingTime>[^\"]+)\")?)   -- matches [-t "estimatedWorkingTime"] if present. Optional
+     *                                                   -- None of the above fields accept " as a valid character.
+     *
+     * (?<invalid>.*)                                    -- matches [invalid]
+     *                                                      Any other excess inputs
+     */
+
+    private static final String ADD_FORMAT = "(task\\s+\\\"(?<taskName>[^\\\"]+)\\\"(\\s+(-m\\s+(?<taskModule>\\w+)))?"
+            + "(\\s+-d\\s+\\\"(?<taskDescription>[^\\\"]+)\\\")?"
+            + "(\\s+-t\\s+\\\"(?<estimatedWorkingTime>[^\\\"]+)\\\")?)(?<invalid>.*)";
```

Comment:
This fix the issue [#177](https://github.com/AY2122S2-CS2113T-T10-3/tp/issues/177)